### PR TITLE
Added support for multi-line metadata

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Added support for multi-line metadata (both parameters & outputs)
- Tested it with an availability set example

Parameter reference
[Compute: AvailabilitySets](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.availabilitysets.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: utilities/tools/Set-ModuleReadMe.ps1 [hunk 1]
```diff
@@ -210,7 +210,7 @@ function Set-ParametersSection {
             # Add external single quotes to all default values of type string except for those using functions
             $defaultValue = ($parameter.defaultValue -is [array]) ? ('[{0}]' -f ($parameter.defaultValue -join ', ')) : (($parameter.defaultValue -is [hashtable]) ? '{object}' : (($parameter.defaultValue -is [string]) -and ($parameter.defaultValue -notmatch '\[\w+\(.*\).*\]') ? '''' + $parameter.defaultValue + '''' : $parameter.defaultValue))
             $allowedValue = ($parameter.allowedValues -is [array]) ? ('[{0}]' -f ($parameter.allowedValues -join ', ')) : (($parameter.allowedValues -is [hashtable]) ? '{object}' : $parameter.allowedValues)
-            $description = $parameter.metadata.description
+            $description = $parameter.metadata.description.Replace("`r`n", '<p>').Replace("`n", '<p>')
```
Comment (eriqua) [inline utilities/tools/Set-ModuleReadMe.ps1:213]: QQ has this been used to regenerate all module readmes? Just to double check for any breaking changes although I don't think we are using multiline descriptions anywhere else. 

Besides that looks great :)

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:213]: The expectation is that no files are updated

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:213]: done

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/readme.md [hunk 2]
```diff
@@ -92,6 +92,6 @@ Create a role assignment for the given resource. If you want to assign a service
 
 ## Template references
 
+- [Define resources with Bicep and ARM templates](https://docs.microsoft.com/en-us/azure/templates)
```
Comment (eriqua) [inline arm/Microsoft.NetApp/netAppAccounts/capacityPools/readme.md:95]: Where does this come from?

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/readme.md [hunk 3]
```diff
@@ -92,6 +92,6 @@ Create a role assignment for the given resource. If you want to assign a service
...
 - [Netappaccounts/Capacitypools](https://docs.microsoft.com/en-us/azure/templates/Microsoft.NetApp/2021-06-01/netAppAccounts/capacityPools)
 - [Netappaccounts/Capacitypools/Volumes](https://docs.microsoft.com/en-us/azure/templates/Microsoft.NetApp/2021-06-01/netAppAccounts/capacityPools/volumes)
-- [Roleassignments](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Authorization/roleAssignments)
```
Comment (eriqua) [inline arm/Microsoft.NetApp/netAppAccounts/capacityPools/readme.md:97]: Why did we loose that?

File: arm/Microsoft.Network/firewallPolicies/deploy.bicep [hunk 4]
```diff
@@ -72,7 +72,7 @@ param fqdns array = []
 @description('Optional. List of IP addresses for the ThreatIntel Allowlist.')
 param ipAddresses array = []
 
-@description('Optional. Secret Id of (base-64 encoded unencrypted pfx) Secret or Certificate object stored in KeyVault.	')
+@description('Optional. Secret ID of (base-64 encoded unencrypted pfx) Secret or Certificate object stored in KeyVault.	')
```
Comment (eriqua) [inline arm/Microsoft.Network/firewallPolicies/deploy.bicep:75]: Shouldn't we get a readme update for this?

File: utilities/tools/Set-ModuleReadMe.ps1 (hunk 1 above)
Comment (eriqua) [reply utilities/tools/Set-ModuleReadMe.ps1:213]: A few files have been updated, but I don't think that was due to the readme generator change, was it? So it's probably due to module changes not including the readme update. How can we avoid that besides checking it while reviewing? Open question not strictly related to this PR

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:213]: No, its a general change. Presumably for modules that were updated before the new script was live / people did not use it

File: arm/Microsoft.Network/firewallPolicies/deploy.bicep (hunk 4 above)
Comment (MrMCake) [reply arm/Microsoft.Network/firewallPolicies/deploy.bicep:75]: On my end it matches

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/readme.md (hunk 3 above)
Comment (MrMCake) [reply arm/Microsoft.NetApp/netAppAccounts/capacityPools/readme.md]: Only god knows. Just ran-it again and it was automatically re-added. I suspect the multi-theading causes bad pings

File: utilities/tools/Set-ModuleReadMe.ps1 (hunk 1 above)
Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:213]: Wouldn't know how else to check it unless we want to write more in-depth readme tests (e.g. do all columns have `''` in it if there is an empty default value, etc.)

File: arm/Microsoft.NetApp/netAppAccounts/capacityPools/readme.md (hunk 2 above)
Comment (MrMCake) [reply arm/Microsoft.NetApp/netAppAccounts/capacityPools/readme.md]: Strnage. Doesn't show up for me (locally)

File: utilities/tools/Set-ModuleReadMe.ps1 (hunk 1 above)
Comment (eriqua) [reply utilities/tools/Set-ModuleReadMe.ps1:213]: or maybe just run the readme generator in the validation step and fail if changes are added ?

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:213]: Feel free to create an issue for it. I guess we'd need to impement some sort of hash-comparison to detect if files were changed

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**